### PR TITLE
Updated API, more customization and improved docs

### PR DESCRIPTION
**Pull request by apalala-dev**
## Description

This PR intends to provide a better documentation, fix bugs and provide an API easier to use.
Among the things I changed, there are:
- Fix for aspect ratio on Android
- Tap to focus on Android (global focus on iOS)
- Pinch and tap on preview customizable with examples (tap to take a picture)
- A few renames and other small fixes for better consistency
- Updated the docs with the above changes and with some illustrations

## Checklist

Before creating any Pull Request, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).

- [x] 📕 I read the [Contributing page](https://github.com/Apparence-io/camera_awesome/blob/master/CONTRIBUTING.md).
- [ ] 🤝 I match the actual coding style.
- [ ] ✅ I ran ```flutter analyze``` without any issues.

## Breaking Change

- [x] 🛠 My feature contain breaking change.

The constructor of CamerAwesome now takes a required AwesomeFileSaver as parameter instead of needing fields without requiring them.

Example:
``` dart
CameraAwesomeBuilder.awesome(
  awesomeFileSaver: AwesomeFileSaver.imageAndVideo(
    imagePathBuilder: () => _path(CaptureModes.PHOTO),
    videoPathBuilder: () => _path(CaptureModes.VIDEO),
    initialCaptureMode: CaptureModes.PHOTO,
  ),
  onMediaTap: (mediaCapture) {
    OpenFile.open(mediaCapture.filePath);
  },
)
```

Several constructors exist depending on if one wants to save pictures, videos or both.

### CODE REVIEW

**Comment by apalala-dev** [comment]
Note: to see the updated docs, get the branch on your local environment and go to https://docs.page/preview

File: example/lib/ai_analysis_ex.dart [hunk 1]
```diff
@@ -72,9 +72,11 @@ class _CameraPageState extends State<CameraPage>
         children: <Widget>[
           Positioned.fill(
             child: CameraAwesomeBuilder.awesome(
-              initialCaptureMode: CaptureModes.PHOTO,
-              picturePathBuilder: (captureMode) => _path(captureMode),
-              videoPathBuilder: (captureMode) => _path(captureMode),
+              awesomeFileSaver: AwesomeFileSaver.imageAndVideo(
+                imagePathBuilder: () => _path(CaptureModes.PHOTO),
+                videoPathBuilder: () => _path(CaptureModes.VIDEO),
+                initialCaptureMode: CaptureModes.PHOTO,
```
Comment (g-apparence) [inline example/lib/ai_analysis_ex.dart:78]: c'est bizarre d'avoir le initialCaptureMode dans le paramètre awesomeFilesaver ?
On pourrait renommer le paramètre awesomeFileSaver en pathBuilders ?

**Review by g-apparence — CHANGES_REQUESTED** [review]

File: example/lib/ai_analysis_ex.dart (hunk 1 above)
Comment (apalala-dev) [reply example/lib/ai_analysis_ex.dart:78]: Renamed into SaveConfig

Comment (Zazo032) [reply example/lib/ai_analysis_ex.dart:78]: Leaving this style recommendation here for the enums: https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names

**Pull request merged**